acceptance/framework/vault/helpers.go

```diff
@@ -80,177 +28,183 @@ func GenerateGossipSecret() (string, error) {
+	// <baseURL>/issue/<roleName>
+	config.CertPath = ConfigurePKICerts(t, vaultClient, config.BaseURL,
+		config.AllowedSubdomain, config.PolicyName, config.KubernetesNamespace,
+		config.DataCenter, config.MaxTTL)
+	// Configure AuthMethodRole that will map the service account name
+	// to the Vault role
+	authMethodRoleConfig := &KubernetesAuthRoleConfiguration{
+		ServiceAccountName:  config.ServiceAccountName,
+		KubernetesNamespace: config.KubernetesNamespace,
+		AuthMethodPath:      config.AuthMethodPath,
+		RoleName:            config.RoleName,
+		PolicyNames:         config.PolicyName,
 	}
-	_, err := vaultClient.Logical().Write("pki/root/generate/internal", params)
-	require.NoError(t, err)
-
-	err = vaultClient.Sys().PutPolicy("consul-ca", caPolicy)
-	require.NoError(t, err)
+	authMethodRoleConfig.ConfigureK8SAuthRole(t, vaultClient)
 }
 
-// ConfigurePKICertificates configures roles in Vault so that Consul server TLS certificates
-// can be issued by Vault.
-func ConfigurePKICertificates(t *testing.T, vaultClient *vapi.Client, consulReleaseName, ns, datacenter string, maxTTL string) string {
-	consulServerDNSName := consulReleaseName + "-consul-server"
-	allowedDomains := fmt.Sprintf("%s.consul,%s,%s.%s,%s.%s.svc", datacenter, consulServerDNSName, consulServerDNSName, ns, consulServerDNSName, ns)
-	params := map[string]interface{}{
-		"allowed_domains":    allowedDomains,
-		"allow_bare_domains": "true",
-		"allow_localhost":    "true",
-		"allow_subdomains":   "true",
-		"generate_lease":     "true",
-		"max_ttl":            maxTTL,
-	}
-
-	pkiRoleName := fmt.Sprintf("server-cert-%s", datacenter)
-
-	_, err := vaultClient.Logical().Write(fmt.Sprintf("pki/roles/%s", pkiRoleName), params)
-	require.NoError(t, err)
-
-	certificateIssuePath := fmt.Sprintf("pki/issue/%s", pkiRoleName)
-	serverTLSPolicy := fmt.Sprintf(`
-path %q {
-  capabilities = ["create", "update"]
-}`, certificateIssuePath)
-
-	// Create the server policy.
-	err = vaultClient.Sys().PutPolicy(pkiRoleName, serverTLSPolicy)
-	require.NoError(t, err)
-
-	return certificateIssuePath
+type SaveVaultSecretConfiguration struct {
+	Path       string
+	Key        string
+	PolicyName string
+	Value      string
 }
 
-// ConfigureACLTokenVaultSecret generates a token secret ID for a given name,
-// stores it in Vault as a secret and configures a policy to access it.
-func ConfigureACLTokenVaultSecret(t *testing.T, vaultClient *vapi.Client, tokenName string) string {
-	// Create the Vault Policy for the token.
-	logger.Logf(t, "Creating %s token policy", tokenName)
-	policyName := fmt.Sprintf("%s-token", tokenName)
-	tokenPolicy := fmt.Sprintf(tokenPolicyTemplate, tokenName)
-	err := vaultClient.Sys().PutPolicy(policyName, tokenPolicy)
-	require.NoError(t, err)
-
-	// Generate the token secret.
-	token, err := uuid.GenerateUUID()
+func (config *SaveVaultSecretConfiguration) Save(t *testing.T, vaultClient *vapi.Client) {
```

Comment:
Could you add some documentation on the `Save` function please?
It is not clear by the name what it does.
My understanding is that it creates a KV2 secret and a policy with read access?
Maybe it could be named something more descriptive 🤷🏽

Reply:
Changed the struct name to `KV2Secret` and changed the function name to `SaveSecretAndAddReadPolicy`. 🙏